solr/core/src/java/org/apache/solr/handler/loader/XMLLoader.java

```diff
@@ -453,7 +454,12 @@ public SolrInputDocument readDoc(XMLStreamReader parser) throws XMLStreamExcepti
               }
               break;
             }
-            doc.addField(name, v);
+            if(!isLabeledChildDoc){
+              // do not add empty data if a labeled child document was added
```

Comment:
Maybe this could be clarified some; I find it confusing.  If at this point we're *not* a labelled child doc, then yes add a value (it won't be empty; yet the comment suggests it is).  Maybe your comment goes in the "else" section?  It may be simpler to read if we avoid the negation -- do an if(isLabeledChildDoc) and else not labelled child doc.  It appears setting isLabeledChildDoc=false is a "reset" action of sorts; maybe say that.

Reply:
I added this since for some odd reason after parsing the child document, the parser would return a string with 2 spaces(the indent) and add that to the field value list.
Yes setting the isLabeledChildDoc=false is a way to reset so the behaviour with non child document values won't change.